plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-transact-account-manager.php

```diff
@@ -0,0 +1,394 @@
+	 */
+	private function update_merchant_account_cache( $merchant_account ) {
+		$cache_key = $this->gateway->testmode ? self::TRANSACT_MERCHANT_ACCOUNT_CACHE_KEY_TEST : self::TRANSACT_MERCHANT_ACCOUNT_CACHE_KEY_LIVE;
+		$expires   = time() + self::TRANSACT_ACCOUNT_CACHE_EXPIRY;
+		update_option(
+			$cache_key,
+			array(
+				'account' => $merchant_account,
+				'expiry'  => $expires,
+			)
+		);
+	}
+
+	/**
+	 * Update the provider account cache.
+	 *
+	 * @param array $provider_account The provider account data.
+	 */
+	private function update_provider_account_cache( $provider_account ) {
+		$cache_key = $this->gateway->testmode ? self::TRANSACT_PROVIDER_ACCOUNT_CACHE_KEY_TEST : self::TRANSACT_PROVIDER_ACCOUNT_CACHE_KEY_LIVE;
+		$expires   = time() + self::TRANSACT_ACCOUNT_CACHE_EXPIRY;
+		update_option(
+			$cache_key,
+			array(
+				'account' => $provider_account,
+				'expiry'  => $expires,
+			)
+		);
+	}
+
+	/**
+	 * Get the merchant account from the database cache.
+	 *
+	 * @return array|null The merchant account data, or null if the cache is
+	 *                    empty or expired.
+	 */
+	private function get_merchant_account_from_cache() {
+		$cache_key        = $this->gateway->testmode ? self::TRANSACT_MERCHANT_ACCOUNT_CACHE_KEY_TEST : self::TRANSACT_MERCHANT_ACCOUNT_CACHE_KEY_LIVE;
+		$merchant_account = get_option( $cache_key, null );
+
+		if ( empty( $merchant_account ) || ( isset( $merchant_account['expiry'] ) && $merchant_account['expiry'] < time() ) ) {
+			return null;
+		}
+
+		return $merchant_account['account'] ?? null;
+	}
+
+	/**
+	 * Get the provider account from the database cache.
+	 *
+	 * @return bool|null The provider account data, or null if the cache is
+	 *                    empty or expired.
+	 */
+	private function get_provider_account_from_cache() {
+		$cache_key        = $this->gateway->testmode ? self::TRANSACT_PROVIDER_ACCOUNT_CACHE_KEY_TEST : self::TRANSACT_PROVIDER_ACCOUNT_CACHE_KEY_LIVE;
+		$provider_account = get_option( $cache_key, null );
+
+		if ( empty( $provider_account ) || ( isset( $provider_account['expiry'] ) && $provider_account['expiry'] < time() ) ) {
+			return null;
+		}
+
+		return $provider_account['account'] ?? null;
+	}
+
+	/**
+	 * Send a request to the Transact platform.
+	 *
+	 * @param string $method The HTTP method to use.
+	 * @param string $endpoint The endpoint to request.
+	 * @param array  $request_body The request body.
+	 *
+	 * @return array|null The API response body, or null if the request fails.
+	 */
+	private function send_transact_api_request( $method, $endpoint, $request_body ) {
+		if ( 'GET' === $method ) {
+			$endpoint .= '?' . http_build_query( $request_body );
+		}
+
+		$response = Jetpack_Connection_Client::wpcom_json_api_request_as_blog(
+			$endpoint,
+			self::WPCOM_PROXY_ENDPOINT_API_VERSION,
+			array(
+				'headers' => array( 'Content-Type' => 'application/json' ),
+				'method'  => $method,
+				'timeout' => 60,
```

Comment:
I would move the timeout value to a constant to make it easier to read/change in the future if needed